handlers/rsa.go

```diff
@@ -0,0 +1,211 @@
+package handlers
+
+import (
+	"crypto"
+	"crypto/dsa"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"net/http"
+	"path/filepath"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/vocdoni/blind-csp/csp"
+	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/db/metadb"
+	"go.vocdoni.io/dvote/log"
+)
+
+var rsaPubKeys = []string{
+	`-----BEGIN PUBLIC KEY-----
+MIIBIDANBgkqhkiG9w0BAQEFAAOCAQ0AMIIBCAKCAQEApc2hU8zulyJzdQE5IPAv
+B2BgveoZmYUmPEjSb4DViBoATK1hlaY8Psp5vj0H0L4tM8AlXRhPQlECibhgccig
+xQFcG7CLXiSAn7c4XoR+J2SCgx76Fwl9L3WhQigxyKsmpGIqubseydmwfJi4TBnq
+qnX4prsW1PT8GpG35t8Qi8PtkXVGmL7G5pkPXtF0hRzKSfhzsDBbJsl6Jk/Rn5Id
+pKHXL22FdbE9fGzIlW2a6Zdd0b0Q3FZBMnWLSwo0OwBtC/qNnDTCzboig9djiFmA
+yuj8jVhsy050nI72TAONjGKi+xn4lYfdOV2k6TyvpRHfylHouK2v0/bktSlkFI0y
+nwIBAw==
+-----END PUBLIC KEY-----`,
+}
+
+// RsaHandler is a handler that allows only 1 registration for IP
+type RsaHandler struct {
+	kv       db.Database
+	keysLock sync.RWMutex
+}
+
+func (rh *RsaHandler) addKey(index, value []byte) {
+	rh.keysLock.Lock()
+	defer rh.keysLock.Unlock()
+	tx := rh.kv.WriteTx()
+	defer tx.Discard()
+	if err := tx.Set(index, value); err != nil {
+		log.Error(err)
+	}
+	if err := tx.Commit(); err != nil {
+		log.Error(err)
+	}
+}
+
+func (rh *RsaHandler) exist(index []byte) bool {
+	rh.keysLock.RLock()
+	defer rh.keysLock.RUnlock()
+	tx := rh.kv.WriteTx()
+	defer tx.Discard()
+	_, err := tx.Get(index)
+	return err == nil
+}
+
+// GetName returns the name of the handler
+func (rh *RsaHandler) GetName() string {
+	return "rsa"
+}
+
+// Init initializes the handler.
+// Takes one argument for persistent data directory.
+func (rh *RsaHandler) Init(opts ...string) (err error) {
+	rh.kv, err = metadb.New(db.TypePebble, filepath.Clean(opts[0]))
+	return err
+}
+
+// Auth is the handler for the rsa handler
+func (rh *RsaHandler) Auth(r *http.Request,
+	ca *csp.Message, pid []byte, st string) (bool, string) {
+	log.Infof(r.UserAgent())
+
+	voterId, message, signature, err := parseRsaAuthData(ca.AuthData)
+	if err != nil {
+		log.Error(err)
```

Comment:
In general its usually a better approach to just return the error (and not printing it) in order to let the layer behind decide how to handle this error (either printing it or doing something else).

Reply:
This applies also to all other cases like this one

Reply:
I followed what I found on the IP Address example (log.Infof, log.Warnf + return string) Happy to skip duplicate stuff 👍